### PR TITLE
Transformations: Grouping to matrix empty value option

### PR DESCRIPTION
**Pull request by hugo082**
Allow grouping matrix data transformer to define the value used when the matrix entry is not defined

**What this PR does / why we need it**: This PR allows us to define the value to use when the matrix entry is not defined. Currently, the transformers always use an empty string as the default value which breaks some visualization options (e.g. _Connect null values_ on _State Timeline_ visualizer)

### Example

**Input data:**
| Time | Station | State |
|--|--|--|
|12:50|A|1|
|12:30|B|1|
|12:10|A|1|

**Output:**
| | A | B |
|--|--|--|
|12:50|1| |
|12:30| |1|
|12:10|1| |

**Expected:** (with `emptyValue = SpecialValue.Null`)
| | A | B |
|--|--|--|
|12:50|1| null |
|12:30| null |1|
|12:10|1| null |

### CODE REVIEW

**Comment by mellieA** [comment]
@hugo082 thank you for contributing! I've added this to the metrics squad project for review

File: packages/grafana-data/src/transformations/transformers/groupingToMatrix.test.ts [hunk 1]
```diff
@@ -105,6 +105,43 @@ describe('Grouping to Matrix', () => {
...
+        {
+          name: 'Time\\Time',
+          type: FieldType.string,
+          values: new ArrayVector([1000]),
+          config: {},
+        },
+        {
+          name: '1000',
+          type: FieldType.number,
+          values: new ArrayVector([1, null]),
```
Comment (bohandley) [inline packages/grafana-data/src/transformations/transformers/groupingToMatrix.test.ts:136]: @kylebrandt This is an example of "No data" right? https://github.com/grafana/grafana-plugin-sdk-go/tree/main/data/contract_docs#no-data-and-empty

**Review by bohandley — DISMISSED** [review]
LGTM! Thank you for your contribution!

**Comment by bohandley** [comment]
Also, @hugo082 I will need to check this in enterprise. May I please have access to your fork? Thank you!

**Comment by bohandley** [comment]
Could you also update your forked version of hugo082:main so that is up to date with grafana:main? I am unable to do the enterprise check. Thank you!

**Comment by hugo082** [comment]
@bohandley Thanks for your review! The main branch is now up to date and the frontend test is fixed ✅

**Comment by bohandley** [comment]
Hi @hugo082! I made the last change necessary on your fork, sorry about the typo! I think it's good to go! The tests are all passing too.

**Review by bohandley — APPROVED** [review]
LGTM!

**Comment by bohandley** [comment]
Enterprise tests are passing
Screen Shot 2022-12-20 at 2 55 29 PM

File: public/app/features/transformers/editors/GroupingToMatrixTransformerEditor.tsx [hunk 2]
```diff
@@ -61,6 +79,9 @@ export const GroupingToMatrixTransformerEditor: React.FC<TransformerUIProps<Grou
         <InlineField label="Cell Value" labelWidth={10}>
           <Select options={fieldNames} value={options.valueField} onChange={onSelectValue} isClearable />
         </InlineField>
+        <InlineField label="Empty Value" labelWidth={10}>
```
[the remark this answers is not in the document]
Comment (bohandley) [reply public/app/features/transformers/editors/GroupingToMatrixTransformerEditor.tsx]: I made the update, it was a little one!

**Pull request merged**